sdk/iot/Azure.Iot.Hub.Service/src/JobsClient.cs

```diff
@@ -3,23 +3,205 @@
+            {
+                Type = JobPropertiesType.Export,
+                OutputBlobContainerUri = outputBlobContainerUri.ToString(),
+                ExcludeKeysInExport = excludeKeys,
+                StorageAuthenticationType = options?.AuthenticationType,
+                OutputBlobName = options?.OutputBlobName
+            };
+
+            return _jobsRestClient.CreateImportExportJob(jobProperties, cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates a job to export device registrations to the container.
+        /// </summary>
+        /// <param name="outputBlobContainerUri">URI containing SAS token to a blob container. This is used to output the results of the export job.</param>
+        /// <param name="excludeKeys">If true, authorization keys are NOT included in export output.</param>
+        /// <param name="options">The optional settings for this request.</param>
+        /// <param name="cancellationToken">Task cancellation token.</param>
+        /// <returns>JobProperties of the newly created job.</returns>
+        public virtual Task<Response<JobProperties>> CreateExportDevicesJobAsync(
+            Uri outputBlobContainerUri,
+            bool excludeKeys,
+            ExportJobRequestOptions options = default,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(outputBlobContainerUri, nameof(outputBlobContainerUri));
+
+            var jobProperties = new JobProperties
+            {
+                Type = JobPropertiesType.Export,
+                OutputBlobContainerUri = outputBlobContainerUri.ToString(),
+                ExcludeKeysInExport = excludeKeys,
+                StorageAuthenticationType = options?.AuthenticationType,
+                OutputBlobName = options?.OutputBlobName
+            };
+
+            return _jobsRestClient.CreateImportExportJobAsync(jobProperties, cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates a job to import device registrations into the IoT Hub.
+        /// </summary>
+        /// <param name="importBlobContainerUri">URI containing SAS token to a blob container that contains registry data to sync.</param>
+        /// <param name="outputBlobContainerUri">URI containing SAS token to a blob container. This is used to output the status of the job.</param>
+        /// <param name="options">The optional settings for this request.</param>
+        /// <param name="cancellationToken">Task cancellation token.</param>
+        /// <returns>JobProperties of the newly created job.</returns>
+        public virtual Response<JobProperties> CreateImportDevicesJob(
+            Uri importBlobContainerUri,
+            Uri outputBlobContainerUri,
+            ImportJobRequestOptions options = default,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(importBlobContainerUri, nameof(importBlobContainerUri));
```

Comment:
Why parameter checks here, but not for export devices job methods?

Reply:
There is parameters checks for both jobs. Can you point me to which method does not have it?

Reply:
I must have been drinking.